### DISCUSSION TITLE
It is still worth to use the original source code?

### DISCUSSION THREAD

**Discussion by skdooboomer** [FunkinCrew/Funkin#1921] [Q&A] [answered]
### **With all these engines that we got, it's still worth to use the old source code?**

The old engine isn't customizable, bad input and it's very buggy. Otherwise, we have Psych Engine that is extremely customizable and easy to code. We also have Kade Engine, that is also good for making advanced stuff and it's also extremely customizable. So, do i use the old source code to make my mods or redo the code or i use a already existing engine to my mods?
![Which to use](https://user-images.githubusercontent.com/94766081/147988221-a67a56af-e205-46ce-8787-562bb861c488.png)

**Comment by libauror-a** [comment, marked as answer]
go with kade engine, please don't use the original engine, it sucks. Plus don't try making your own engine. (like i'm doing)

**Reply by skdooboomer** [reply]
> go with kade engine, please don't use the original engine, it sucks. Plus don't try making your own engine. (like i'm doing)

Thanks!

**Reply by DeceptedPar** [reply]
kade engine is good for when it comes to modchart compatibility and such but it has problem for when it comes to lag but if you want an experience that's more user friendly and has built in stuff like character editor week editor, go with psych it's your choice as I can't and wont force you to pick an engine you don't want to use

**Comment by ghost** [comment]
Kade is heavily modified, and Psych literally doesn't need code. So yes I would say it is still worth it to use the OG source.

You can copy the input system (which isn't too hard) and code different events and animations. Although I'm not gonna recommend it because @ThePlank is sort of right. Kade is heavily modified, so you can see it as easier to make mods with. And Psych? You don't even need the source code because almost every building aspect in the game can be softcoded.

Also, there are more engines like Forever Engine, Andromeda Engine, Mic'd Up Engine and countless others. So it really doesn't matter what you choose, because most source codes from engines are similar.

**Comment by NokioPyro** [comment]
Personally I would use Psych, Its more simple for beginners, But sadly its mainly lua thats talked about in there,
kade is just a more advanced one, I couldnt even make a stage in it bloody hell
You can use FPSPlus, But the ghost input sucks ass. Project fnf is a good engine, I dont know about any flaws in it.
And Modding + I have no clue about it.

**Reply by Johferson** [reply]
FPS Plus just has bad inputs. I know so because I played Kolsan's HD mod (which is the Week 4 mod update) and boy do I need to explain how I always die in M.I.L.F even though I always FC the vanilla one in Kade.

**Comment by DabbinDude316B** [comment]
Go on kade if you feel like it, but pysch engine is easier for modding.
